### ISSUE TITLE
Adding compass and scale_bar only once in `tm_facets`

### ISSUE THREAD

**Issue by bappa10085** [r-tmap/tmap#837]
I want to add compass and scale_bar only once while using `tm_facets`. Here is an example code
```
library(tmap)
library(stars)
library(terra)

#Read some data
tif = system.file("tif/L7_ETMs.tif", package = "stars")
x = rast(tif)

#Plot it using tmap r package
tm_shape(x) +
  tm_raster(style="quantile")+
  tm_facets(nrow = 3) +
  tm_layout(panel.labels = names(x), 
            legend.outside=T, legend.outside.position = "right",
            legend.position= c("center", "center"),
            legend.text.size = 1,
            legend.format = list(digits = 2, text.separator = "-"))+
  tm_compass(position = c("RIGHT", "BOTTOM"))+
  tm_scale_bar(text.size = 1, position = c("RIGHT", "BOTTOM"))
```
![Rplot](https://github.com/r-tmap/tmap/assets/17943205/486242ab-f761-403c-b359-b45132516b56)

As you can see in the output, all the facets are having compass and scale_bar. How can I add compass and scale_bar in only one facet?

**Comment by aonojeghuo** [comment]
Hi @bappa10085, you can try this:

    #Read some data
    tif = system.file("tif/L7_ETMs.tif", package = "stars")
    x = terra::rast(tif)

    #Plot it using tmap r package
    tm_shape(x) +    tm_raster(style="quantile")+    tm_facets(nrow = 3) +
     tm_layout(panel.labels = names(x), attr.outside = T,attr.outside.position = "bottom" ,attr.just="right",
            legend.outside=T, legend.outside.position = "right",
            legend.position= c("left", "top"),
            legend.text.size = 0.8,legend.title.size = 0.9,
            legend.format = list(digits = 2, text.separator = "-"))+
    tm_compass(position = c("left", "top"))+
    tm_scale_bar(text.size = 1, position = c("RIGHT", "top"))

I cant figue out how to move it to the right ourside despite defining it but you now have just 1 scale bar and north arrow

![image](https://github.com/r-tmap/tmap/assets/30529377/15f81f2b-74b4-49f5-b456-c54682bb4f66)

**Comment by andrewmaclachlan** [comment]
Hi there, i am struggling to replicate this in the latest version of tmap (4.1). The code produces a facet plot with a legend, scale bar and N arrow per facet. I have resolved the legend with fill.free=FALSE but i can't get a shared scale bar or N arrow. Any ideas would be welcomed.

**Comment by andrewmaclachlan** [comment]
Thanks @Nowosad! This might be a dumb question but is there a limit on the number of items that can be positioned outside the main plot. For example, if i add `tm_credits()` or `tm_title()` i get the following error: 

`Error in comp$units$unit : $ operator is invalid for atomic vectors`. 

Code

`library(tmap)
library(terra)

tif = system.file("tif/L7_ETMs.tif", package = "stars")
x = terra::rast(tif)

tm_shape(x) +   
  tm_raster(col.scale = tm_scale_intervals(style = "quantile"),
            col.free = FALSE,
            col.legend = tm_legend(position = tm_pos_out())) +  
  tm_facets(nrow = 3) +
  tm_layout()+
  tm_compass(position = tm_pos_out("center", "bottom", pos.h = "left"))+
  tm_scalebar(text.size = 1, position = tm_pos_out("center", "bottom", pos.h = "right"))+
  tm_credits("credit test", position = tm_pos_out("center", "top", pos.h = "left"))`

**Comment by Nowosad** [comment]
@andrewmaclachlan -- please open a new issue. It seems like a bug.
